### DISCUSSION TITLE
Static analysis by default and webpack replacement

### DISCUSSION THREAD

**Discussion by Katletos** [h5bp/html5-boilerplate#3149] [Ideas]
Hi, i have 2 proposals: 
- What do you think about adding ESLint by default to maintain code quality? In production, linters have become an industry standard. Why not bring it into the project?
- There is now a new generation of build systems (ESBuild, Vite) that are faster, easier to learn and do many things out of the box: code minimisation, etc. Might it be advisable to switch to it?

If this discussion is supported, I will implement this in whole (ESBuild + ESLint) or in part (ESLint)

**Comment by roblarsen** [comment]
If these new build systems can do things like... Copy static files without them being referenced in JavaScript (shocking!) and not cache-bust favicon.ico (madness!) then I'd at least consider it. I looked at Vite and it seemed to fail. This is not solely a JavaScript project so a lot of these build systems are too simplistic and opinionated for our purposes. New is not automatically better.

**Comment by roblarsen** [comment]
#2487 #2474 #2876 (parcel is dumb) #2568

**Comment by roblarsen** [comment]
@Katletos on the flip side, shipping eslint with the project seems okay.

**Comment by Katletos** [comment]
@roblarsen maybe we should also add css minification and image lossless compression by default?
